x/mint/client/rest/grpc_query_test.go

```diff
@@ -0,0 +1,114 @@
+package rest_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/testutil"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	grpctypes "github.com/cosmos/cosmos-sdk/types/grpc"
+	"github.com/gogo/protobuf/proto"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/cosmos/cosmos-sdk/testutil/network"
+	minttypes "github.com/cosmos/cosmos-sdk/x/mint/types"
+)
+
+type IntegrationTestSuite struct {
+	suite.Suite
+	cfg     network.Config
+	network *network.Network
+}
+
+func (s *IntegrationTestSuite) SetupSuite() {
+	s.T().Log("setting up integration test suite")
+
+	cfg := network.DefaultConfig()
+
+	genesisState := cfg.GenesisState
+	cfg.NumValidators = 1
+
+	var mintData minttypes.GenesisState
+	s.Require().NoError(cfg.Codec.UnmarshalJSON(genesisState[minttypes.ModuleName], &mintData))
+
+	inflation := sdk.MustNewDecFromStr("1.0")
+	mintData.Minter.Inflation = inflation
+	mintData.Params.InflationMin = inflation
+	mintData.Params.InflationMax = inflation
+
+	mintDataBz, err := cfg.Codec.MarshalJSON(&mintData)
+	s.Require().NoError(err)
+	genesisState[minttypes.ModuleName] = mintDataBz
+	cfg.GenesisState = genesisState
+
+	s.cfg = cfg
+	s.network = network.New(s.T(), cfg)
+
+	_, err = s.network.WaitForHeight(1)
+	s.Require().NoError(err)
+}
+
+func (s *IntegrationTestSuite) TearDownSuite() {
+	s.T().Log("tearing down integration test suite")
+	s.network.Cleanup()
+}
+
+func (s *IntegrationTestSuite) TestQueryGRPC() {
+	val := s.network.Validators[0]
+	baseURL := val.APIAddress
+	testCases := []struct {
+		name     string
+		url      string
+		headers  map[string]string
+		respType proto.Message
+		expected proto.Message
+	}{
+		{
+			"gRPC request params",
+			fmt.Sprintf("%s/cosmos/mint/v1beta1/params", baseURL),
+			map[string]string{
+				grpctypes.GRPCBlockHeightHeader: "1",
+			},
```

Comment:
Since we're not really testing height-header-related logic, I would remove it from the tests, to make it less distracting for readers.

Also see https://github.com/cosmos/cosmos-sdk/pull/7173#discussion_r482969324